concepts/Enarx_Public_Chat_App_Zulip/000005-Enarx_Public_Chat_App_Zulip.md

```diff
@@ -0,0 +1,126 @@
+
+Zulip, on the other hand, requires a login, which presents a barrier to people being able to see the discussions going on about Enarx. An idle observer would need to create an account, which they might not want to do.
+
+There is a potential workaround present, however, with [`zulip-archive`](https://github.com/zulip/zulip-archive). This is a tool, runnable as a Github Action, that archives our chat logs and publishes them on a publicly-accessible Github Pages site. For Enarx, this may manifest itself as a sub-link of `enarx.github.io` -- for example, we may be able to point people to `enarx.github.io/zulip-archive` for a publicly-accessible chat record.
+
+In terms of implementation, it seems fairly straightforward, and there are examples of this being done on an organizational level (see [here](https://leanprover-community.github.io/archive/)). An Enarx deployment of this tool would look fairly similar. We would need to create a new repository within the Enarx organization to hold the chat record, though it shouldn't need to be maintained after `zulip-archive` is set up.
+
+## Rationale and alternatives
+
+We think Zulip is the superior choice due to a number of feature advantages it has over Gitter and other solutions.
+- Organization
+    - Zulip has topic-based organization, allowing better separation of thoughts and tasks than Gitter, which only allows for a chronological feed.
+    - Like Gitter, it supports multiple rooms (known as “streams”). It also promotes stream discoverability better than Gitter.
+    - Zulip supports chat search, where Gitter offers nothing of the sort.
+- App support (especially mobile)
+    - Gitter’s mobile app is, to put it kindly, not very good. It doesn’t support notifications at all, it is slow, and it doesn’t register read messages well. Additionally, Gitter themselves do not officially recommend their own apps, and [say](https://gitter.im/apps) they may be deprecated in the future. Zulip’s mobile apps work well, do support notifications, and generally support a larger subset of features.
+    - Both have desktop Linux apps available, though both are also available in browser.
+    - Both have Windows and Mac desktop clients.
+- Nice-to-have features
+    - Zulip supports file uploads, where Gitter does not. This can be helpful if we need to share a screenshot, for example.
+    - Zulip allows users to link to a unique message in the chat.
+    - Zulip appears to have much more robust integration support, including Github integration. Gitter’s Github integration is out of date and does not work.
+
+In terms of alternatives: Following Enarx's design principles of open development, we strongly value FOSS solutions, of which there are, unfortunately, few. Beyond Zulip, there's:
+
+- Gitter, which we've already discussed
+- Matrix, which has some nice security/privacy features, but otherwise looks high-maintenance and not necessarily as feature-complete as Zulip.
```

Comment:
```suggestion
- [Matrix](https://matrix.org), which has some nice security/privacy features, but otherwise looks high-maintenance and not necessarily as feature-complete as Zulip.
```

Reply:
I'm not sure what you mean by high-maintenance. The 3 options right now for using matrix as a chat system are:
1. public servers: create accounts on a public server (ie: matrix.org)
2. self-hosted: run your own server and create accounts there
3. paid hosting: this is [what Mozilla is doing](https://chat.mozilla.org/#/welcome), using modular.im

1 and 3 require no effort, but 3 requires money. 2 requires effort.

Threaded conversations and custom emojis are not available in any Matrix client right now, and "communities" (groups of channels/chats) are in need of more work to make them nice to use (which is planned) otherwise, i think everything else is on par with Zulip, or superior (good bridging to multiple other networks, end-to-end encryption).

I don't want to make a big case for Matrix, as i think the end-user experience is still a little less polished than Zulip's (even though that will change in the near future).

But it is decentralised, which means that the conversation can keep going when a server goes down (although users on that server are obviously disconnected).

Reply:
All fair points. I've amended the section to be more accurate (in truth, I have little experience with Matrix proper, so your feedback is valuable).